packages/gatsby-plugin-sharp/src/index.js

```diff
@@ -464,13 +176,29 @@ function queueImageResizing({ file, args = {}, reporter }) {
   const prefixedSrc =
     options.pathPrefix + `/static/${digestDirPrefix}` + encodedImgSrc
 
+  // Create job and add it to the queue, the queue will be processed inside gatsby-node.js
+  const job = {
+    file,
+    args: options,
+    inputPath: file.absolutePath,
+    outputPath: filePath,
+    src: prefixedSrc,
+  }
+
+  queue.set(prefixedSrc, job)
+
   return {
     src: prefixedSrc,
     absolutePath: filePath,
     width,
     height,
     aspectRatio,
-    finishedPromise,
+    // finishedPromise is needed to not break our API (https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-transformer-sqip/src/extend-node-type.js#L115)
+    finishedPromise: () => {
```

Comment:
Will this actually work? Maybe we should make lazy image generation more opt-in? So `gatsby-transformer-sharp` and `gatsby-remark-images` specifically opt-in into it with something like extra `lazy` flag? and keep current behaviour of scheduling job asap if caller didn't opt into it?

Reply:
I'm checking if I don't break but I actually like this approach as no one needs to opt-in for speed. Most plugins have enough with just queuing a job.

I probably write an  RFC about cleaning this plugin up, it will need some changes to core to expose reporter and createJob from a require.

Reply:
@pieh I added a test to demonstrate that this is working. I also tested it with the examples/using-sqip where this is used and it is working